x/upgrade/types/plan.go

```diff
@@ -72,6 +72,11 @@ func (p Plan) DueAt() string {
 	return fmt.Sprintf("height: %d", p.Height)
 }
 
+// IsIBCPlan will return true if plan includes IBC client information
+func (p Plan) IsIBCPlan() bool {
+	return p.UpgradedClientState != nil
```

Comment:
shouldn't consensus state also be non nil? Maybe add a comment why it doesn't check that?

Reply:
Validate makes sure if one is non-nil then both are non-nil

Reply:
Consensus state is not included in the plan because proposal submitter cannot know it in advance. If plan is IBC plan, then upgrade module will set consensus state at last height when its needed